test/engine/update.result

```diff
@@ -1210,3 +1210,195 @@ t:update({{'=', 'field3', 3}})
 s:drop()
 ---
 ...
+--
+-- gh-8226: allow several updates of the same field
+--
+s = box.schema.create_space('test', {engine = engine})
```

Comment:
Moved tests to `box/update.test.lua`.

Not sure I need to add tests testing first field or field in the middle. New related code seems to be orthogonal to position in the array.

May be we can change tests to update field in the middle to check that last field in not corrupted/overwritten somehow? Not sure I need to test first/middle/last element for every case

Reply:
> Not sure I need to add tests testing first field or field in the middle. New related code seems to be orthogonal to position in the array.

I wouldn't say so. There is at least one commit where you touch the tail update data in a struct, which in your tests is almost always empty, it seems.

> May be we can change tests to update field in the middle to check that last field in not corrupted/overwritten somehow?

You could reuse the big complex tuples in `box/update.test.lua` (lines 297, 514) created specifically for that. The tests would remain the same, but would operate on a more complex tuple.

Reply:
> I wouldn't say so. There is at least one commit where you touch the tail update data in a struct, which in your tests is almost always empty, it seems.

If you are referencing commit `update: make '=' NOP in update tree instead of scalar` - this should be covered by existing tests.

> You could reuse the big complex tuples in box/update.test.lua (lines 297, 514) created specifically for that. The tests would remain the same, but would operate on a more complex tuple.

I would say large tuple will make it harder to follow testing case. In the above places it justified as JSON path is tested.

If I use a complex tuple should I in every test case pick up a different place? This seems a bit a different test flavor, more like a random test. If it is a desired then I'm OK to change but it looks a bit of work)

I guess it is not justified currently that there is a field before updated and there is no field after it. It is a remnant of the first version where I used `space:update()`. May be I should just add an after field?

Reply:
Adding an after field would be good. And also keep the existing tests updating the last field as well. For such complicated code there can't be too many tests.

> every test case pick up a different place?

If you want. I would just recommend testing non-last and last field updates.

> but it looks a bit of work)

This isn't much more complicated than your current tests really. You just have to check results one time in the .result file.

Anyway, I will leave that to the next reviewer.

Reply:
Added last field for tuples in test. Typical tuple now looks like this `t = box.tuple.new({7, {a = 1}, 11}) `.